File: Spring/42scj-cerveja-master/src/main/java/com/fiap/cerveja/service/CervejaService.java
package com.fiap.cerveja.service;

import com.fiap.cerveja.domain.Tipo;
import com.fiap.cerveja.dto.CervejaCreateOrUpdateDTO;
import com.fiap.cerveja.dto.CervejaDTO;
import com.fiap.cerveja.dto.CervejaVencimentoDTO;

import java.util.List;

public interface CervejaService {

    List<CervejaDTO> getAll(Tipo tipo);
    CervejaDTO getById(Long id);
    CervejaDTO create(CervejaCreateOrUpdateDTO createOrUpdateDTO);
    CervejaDTO update(Long id, CervejaCreateOrUpdateDTO cervejaCreateOrUpdateDTO);
    CervejaDTO updateVencimento(Long id, CervejaVencimentoDTO cervejaVencimentoDTO);
    void delete(Long id);

}

File: Spring/42scj-cerveja-master/42scj-cerveja-master/cerveja-webflux/src/main/kotlin/br/com/fiap/cervejawebflux/service/CervejaService.kt
package br.com.fiap.cervejawebflux.service

import br.com.fiap.cervejawebflux.dto.CervejaCreateDTO
import br.com.fiap.cervejawebflux.dto.CervejaDTO
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface CervejaService {

    fun findAll(): Flux<CervejaDTO>

    fun create(cervejaCreateDTO: CervejaCreateDTO): Mono<CervejaDTO>

}